src/GafferArnoldUI/ArnoldLightVisualiser.cpp

```diff
@@ -324,6 +337,65 @@ typedef IECorePreview::LRUCache<IECore::MurmurHash, CompoundDataPtr, IECorePrevi
 // Cache cost is in bytes
 SurfaceTextureCache g_surfaceTextureCache( surfaceTextureGetter, 1024 * 1024 * 64 );
 
+//////////////////////////////////////////////////////////////////////////
+// IESVisualisation helpers
+//////////////////////////////////////////////////////////////////////////
+
+IECoreGL::RenderablePtr iesVisualisation( const std::string &filename )
+{
+
+#if AI_VERSION_ARCH_NUM < 6
+	return nullptr;
+#else
+
+	// It's not entirely clear from rendered results exactly how radius
+	// interacts with the profile, so we just draw the normalised distribution
+	// of the profile.
+
+	const AtString f( filename.c_str() );
+
+	const unsigned int w = 64;
+	const unsigned int h = 32;
+
+	float maxIntensity = 0.0f;
+	std::unique_ptr<float[]> iesIntensities( new float[ w * h ] );
+
+	if( !AiLightIESLoad( f, w, h, &maxIntensity, iesIntensities.get() ) )
+	{
+		return nullptr;
+	}
+
+	V3fVectorDataPtr pData = new V3fVectorData;
+	std::vector<V3f> &p = pData->writable();
+
+	for( unsigned int x = 0; x < w; ++x )
+	{
+		for( unsigned int y = 0; y < h; ++y )
+		{
+			const float theta = 2.0f * M_PI * ( float(x) / w );
+			const float phi = M_PI * ( float(y) / h );
+			const unsigned int i = y * w + x;
+
+			if( iesIntensities[i] > 0.0f )
+			{
+				p.push_back(
+					iesIntensities[i] * V3f(
+						sin( phi ) * cos( theta ),
+						sin( phi ) * sin( theta ),
+						cos( phi )
+					)
+				);
+			}
+		}
+	}
+
+	IECoreScene::PointsPrimitivePtr points = new IECoreScene::PointsPrimitive( pData );
+	ToGLPointsConverterPtr pointsConverter = new ToGLPointsConverter( points );
+	return runTimeCast<IECoreGL::Renderable>( pointsConverter->convert() );
```

Comment:
This is making an `IECoreGL::PointsPrimitive` with [type==Disk](https://github.com/ImageEngine/cortex/blob/master/include/IECoreGL/PointsPrimitive.h#L55), which means that it draws as giant disks if you turn off the `Drawing->Points Primitives->Use GL Points` toggle in the Viewer. ToGLPointsConverter is kindof getting in the way here, so let's just construct `IECoreGL::PointsPrimitive( PointsPrimitive::Point )` directly instead.

Reply:
Ahh good to know, thanks, and avoids the conversion, fixed in https://github.com/GafferHQ/gaffer/pull/3531/commits/1bd6edbf77d7a826ada4560a719a6b28ea3b4950.